lib/broadway_cloud_pub_sub/options.ex

```diff
@@ -13,7 +13,7 @@ defmodule BroadwayCloudPubSub.Options do
     # Handled by Broadway.
     broadway: [type: :any, doc: false],
     client: [
-      type: :atom,
+      type: {:or, [:atom, :mod_arg]},
```

Comment:
Allow configuration via:

```elixir
client: {BroadwayCloudPubSub.PullClient, [max_retries: 3, retry_delay_ms: 500, retry_codes: 502]}
```

It is not as clear to me yet how to test this part.

Reply:
Side affect is that this also allows users to pass options to their custom clients.

Reply:
Test asserts that broadway starts, now sure how to test that options are passed to the client.